lucene/core/src/java/org/apache/lucene/analysis/CharArrayMap.java

```diff
@@ -523,6 +523,7 @@ public void clear() {
    * @throws NullPointerException
    *           if the given map is <code>null</code>.
    */
+  @SuppressWarnings("ReferenceEquality")
```

Comment:
This was the example you wanted to try out on how to suppress legitimate warnings of ReferenceEquality ( or any other warnings ) when we start enabling the checks ?

Reply:
Yes, exactly. I wanted the pr to include an example of how this can be done.